virttest/libvirt_vm.py

```diff
@@ -879,6 +913,31 @@ def add_kernel(help_text, cmdline, kernel_path=None, initrd_path=None,
             if maxmemory:
                 numa_memory = int(maxmemory)
             virt_install_cmd += add_numa(numa_vcpus, numa_memory, numa_nodes)
+            if params.get("numa_pin", "no") == "yes":
+                # Get online host numa nodes
+                host_numa_node = utils_misc.NumaInfo()
+                host_numa_node_list = host_numa_node.online_nodes
+                # check if memory is available in host numa node
```

Comment:
@balamuruhans `host_numa_node.online_nodes_withmem` can be used either in from https://github.com/avocado-framework/avocado-vt/pull/1158 or https://github.com/avocado-framework/avocado-vt/pull/1134
which will save us some of computation here.

and regarding hugepage free I hope it should be skipped with separate error instead to club with normal memory.

Reply:
> @balamuruhans host_numa_node.online_nodes_withmem can be used either in from #1158 or #1134
> which will save us some of computation here.
> 
the computation remains to be same with or without this as we need to anyway check the HugePages_Free, so lets have this.

> and regarding hugepage free I hope it should be skipped with separate error instead to club with normal memory.

when we use HugePageConfig, it will try to allocate the memory we request, but if no of HP couldn't be allocated by kernel, then we could get appropriate error in virt-install as it couldn't back the VM,  libvirt_vm will raise it.

Reply:
okies